src/server/components/lists/searches/helpers/getDbServiceTypeFromParameter.ts

```diff
@@ -0,0 +1,11 @@
+import { ServiceType } from "shared/types";
+
+export function getDbServiceTypeFromParameter(param: string) {
+  const normalisedParameterToServiceType: { [key: string]: ServiceType } = {
```

Comment:
Why use `{ [key: string]: ServiceType }` here instead of `Record<string, ServiceType>`

Reply:
`Record` has (or should have!) very specific functionality, I generally prefer to use `Record` when I want to say "this is an object, I know what the keys are, and I want all the keys to be implemented". 

e.g. 

```
const foo: Record<ServiceType, String> = {
  // warning that I haven't implemented all serviceTypes. IDE will also prompt to auto-fill this for me, since it knows that they keys should be
}
```

and whenever a `ServiceType` is removed or added, I should get warnings/errors saying all the Records<ServiceType, T> have not been implemented properly. 

With `{ [key: string]: T }` - my intention is that "it's just an object keyed with strings, and the values are T"

Typescript seems to treat Record<string, T> as valid which imo it shouldn't.. 

Record does this under the hood: 
```
type Record<K extends keyof any, T> = {
    [P in K]: T;
};
```

so would it make sense for K to be string? What is "P in K" if K is string? Compared to P in ServiceType? 

Record<string, T> is treated like 

```
type Foo<T> = {
  [prop: string]: T
}
```

rather than 
```
type Foo<T> = {
 [prop in string]: T // i.e. "iterate through the props of string, and make a mapped type with it"
```

unless I'm missing something "special" about strings and their properties..!